packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -708,6 +708,7 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin,
 
   void _setTextEditingValue(TextEditingValue newValue, SelectionChangedCause cause) {
     textSelectionDelegate.userUpdateTextEditingValue(newValue, cause);
+    selection = textSelectionDelegate.textEditingValue.selection;
```

Comment:
I think instead of making this change we can just use `editableText.textEditingValue` in `text_selection.dart`.